src/sherpa_ai/memory/belief.py

```diff
@@ -174,12 +178,46 @@ def get_action(self, action_name) -> BaseAction:
                 break
         return result
 
+    def get_dict(self):
+        return self.dict
+
+    def get(self, key):
+        """
+        Get value from the dict, the key can be a dot separated string if the value is nested
+        """
+        return pydash.get(self.dict, key)
+
+    def get_all_keys(self):
+        def get_all_keys(d, parent_key=""):
```

Comment:
is there any particular reason we want the get_all_keys function is nested rather than taking it out since there is no other time its being called inside ?